### DISCUSSION TITLE
Loading in RZ 2-D case

### DISCUSSION THREAD

**Discussion by sidharthsarmah** [idaholab/moose#27507] [Q&A Modules: Solid mechanics]
Hi @GiudGiud , I want to do perform a tensile test of a specimen as shown.
<img width="641" alt="image" src="https://github.com/idaholab/moose/assets/110927986/7140b1a2-7d9e-43ff-82ba-a0f2ff985ad4">
1. I want to apply a load on the end face, how do I apply a point load on the face, but it is basically an average the entire area.
2. Is it enough in MOOSE the just change the coordinate to RZ type for changing the 3D to 2D as shown? How does MOOSE know which is the R or Z direction?
<img width="361" alt="image" src="https://github.com/idaholab/moose/assets/110927986/dc4a2d44-f29a-4262-a9ae-d5613408d523">

```
[Problem]
  type = FEProblem
  coord_type = RZ
[]

**Comment by GiudGiud** [comment]
Hello

1) You ll want to apply the load on a sideset then. Depending on the object you use, you will likely specify the local value of the tension, so the average

2) The coord_type is actually in the Mesh block now. 
It should be enough to provide a 2D mesh.
Note that you do not need the entire mesh, half is enough because of the symmetry

Guillaume

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , thanks:
I have modelled a quarter of the sample as shown,
1. Let's say in 3D, I have a force F applied to the area of the face A, what should be the value of F1 that is should apply to the node in 2D case is it:
a) F1=F/A or b) F1=F/4A (since I am considering a quarter area?

2. Is the direction R and Z implicit in this figure correct?
3. 
![image](https://github.com/idaholab/moose/assets/110927986/24cc1de9-cdb9-4f90-8c66-8c8be3581604)

**Reply by GiudGiud** [reply]
F/A should be correct, as long as the drawing is correct (so A is the area on only one side, and F is applied equally on both sides)

No the directions are not implciitly this way, you specify them using a Mesh parameter to make the Z axis the X axis it seems looking at your mesh orientation in the figure

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , yes that's correct, thanks.
Is this correct?
```
[Mesh]
  [./mesh]
    type = FileMeshGenerator
    file = Mesh.inp
    coord_type = RZ
    rz_coord_axis = X
  [../]
[]
```

**Reply by GiudGiud** [reply]
no it's not. Arent you getting unused parameters warnings on the console for this? (maybe not if you are running in mesh-only mode I guess)
It should be
```
[Mesh]
  coord_type = RZ
  rz_coord_axis = X
  [load_mesh]
    type = FileMeshGenerator
    file = Mesh.inp
  []
[]
```

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , thanks for the correction.

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , one more thing, I want to impose a switching BC.
1. The surface of the sample indicated by the red outline, has conc, C=0.127 for time t=10s, after that the conc on the surface is C=0;
2. Similarly till t=10s, the force, F1=0 and once t>10s, F1=0.01*t, is these syntax correct?
<img width="812" alt="image" src="https://github.com/idaholab/moose/assets/110927986/ea06f95d-d561-46c2-90f6-cae12441017e">

```
[NodalKernels]
  [./force_z]
    type = UserForcingFunctionNodalKernel
    variable = disp_x
    boundary = 'right'
    function = 'if(t<10, 0, t*0.001)'
  [../]
[]

  [./s_conc]
    type = ADFunctionDirichletBC
    variable = cL
    boundary = 'Hsurface'
    function = 'if(t<10, 0.127, 0)'
  [../]
```

**Reply by GiudGiud** [reply]
this looks fine to me, if it gets parsed. I usually write ParsedFunctions but I think these expressions in the input will be parsed

**Reply by GiudGiud** [reply]
though in the post you say 

> Similarly till t=10s, the force, F1=0 and once t>10s, F1=0.01*t, is these syntax correct?

but in the input it's t*0.001 not 0.01

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , is this correct?
```
[NodalKernels]
  [./force_z]
    type = UserForcingFunctionNodalKernel
    variable = disp_x
    boundary = 'right'
    function = 'force'
  [../]
[]

  [./s_conc]
    type = ADFunctionDirichletBC
    variable = cL
    boundary = 'Hsurface'
    function = 'conc'
  [../]

[Materials]
  [./force_cond]
    type = ADParsedMaterial
    expression = 'if(t<10, 0, t*0.001)'
    property_name = 'force'
  [../]

  [./conc_cond]
    type = ADParsedMaterial
    expression = 'if(t<10, 0.127, 0)'
    property_name = 'conc'
  [../]
```

**Reply by GiudGiud** [reply]
this looks good to me

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud @bwspenc , I am having problem pin applying a displacement BC to the end face. The examples in MOOSE has boundary defined as `front back` etc, because of which I am a bit confuse on whether I should apply to a point or entire point or how do I apply to the top cylinder.

<img width="447" alt="image" src="https://github.com/idaholab/moose/assets/110927986/0daabcd9-1c35-4203-b131-6f91327a5edb">
<img width="683" alt="image" src="https://github.com/idaholab/moose/assets/110927986/dfea34bf-9719-4913-853d-40ec4533aa34">

The way it is done in ABAQUS is define a reference point for the entire top portion something like this, how do I impose the same BC in MOOSE for a 2D case:
<img width="195" alt="image" src="https://github.com/idaholab/moose/assets/110927986/6e8d3698-463b-4a0f-bbf9-6c834743fee4">

**Comment by GiudGiud** [comment]
Hello

Where it applies is really defined by the sideset / node sets in use.

So maybe send me your mesh to check where it s being applied?

Guillaume

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , here it is, I applied it to a node ID 95.
I applied Disp BC to boundary `Load_x`.
[
[Mesh.txt](https://github.com/idaholab/moose/files/15182676/Mesh.txt)
](url)

**Reply by GiudGiud** [reply]
how do you read a .txt mesh in moose? do we support that?

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , its.inp, .txt is just uploaded for you to check, you can just change it to .inp else here it is.
[Mesh.inp.zip](https://github.com/idaholab/moose/files/15187881/Mesh.inp.zip)

**Reply by GiudGiud** [reply]
boundary 'right' does not exist in your mesh. 
I see 'load_x' but that's only a single node so it wont do if you are trying to apply the force on the entire surface. 

I suggest you re-create 'right' using a mesh generator. 
SidesetFromNormalsGenerator with a normal of '1 0 0' will do it for example. 
https://mooseframework.inl.gov/source/meshgenerators/SideSetsFromNormalsGenerator.html

**Reply by GiudGiud** [reply]
@bwspenc are DirichletBC the right way to impose symmetry boundary conditions?
I would expect something like 0 flux ? maybe 0 normal stress?

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , I have changed the variables to r and z, can you suggest if this is correct?
```
[Mesh]
  [./Load_mesh]
    type = FileMeshGenerator
    file = Mesh.inp
  [../]
[]
[Problem]
  coord_type = RZ
[]
```
```
[BCs]
  [./x_left]
    type = ADDirichletBC
    variable = disp_z
    boundary = 'Sym_x'
    value = 0  
  [../]
  [./y_bottom]
    type = ADDirichletBC
    variable = disp_r
    boundary = 'Sym_y'
    value = 0  
  [../]
  [./x_load]
    type = ADFunctionDirichletBC
    variable = disp_z
    boundary = 'Load_x'
    function = t*0.003
  [../]
[]
```

**Reply by GiudGiud** [reply]
Changing the names of the variables changes nothing solve-wise
the coord_type is a Mesh parameter, it's deprecated to have it in Problem

have you found a tensile test example in MOOSE by any chance?

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , the disp_x and disp_y does not work with `Coord_axis = x`, but the disp_r and disp_z work.
Could you suggest if I can model this bending test for the cylindrical sample as shown in 2D. 
<img width="385" alt="image" src="https://github.com/idaholab/moose/assets/110927986/1b51ad39-8c80-4345-9f9e-42963eaffe22">

I don't think the symmetry will work here as its bending. My doubt is since the load F is applied to only the top portion, can I model it for 2D case, as there is no `theta` to prescribe.

Can you suggest the 2D model on the left, gives which scenario on the right A or B:

<img width="967" alt="image" src="https://github.com/idaholab/moose/assets/110927986/a081e32b-91dc-49c8-b51e-5dc90ca25c66">

How do I get scenario A to be implemented in 2D case with RZ coord?

**Reply by GiudGiud** [reply]
hello

the case on the left is not symmetric in any way. Not rotationally and not with the other plane.
It's time to make a 3D model

Guillaume

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud, I will first try to model the tensile test in the 3D half of the specimen before bending. 

Can you please suggest if there is a concept of a reference point coupling of the clamped portion as shown in fig for ABAQUS. This is because the top portion is clamped, 
Do we need it in MOOSE or we can just prescribed disp BC in all the nodes at the clamped portion?

![image](https://github.com/idaholab/moose/assets/110927986/da27ce39-03f7-4164-9862-2d135eabb8dc)

**Reply by GiudGiud** [reply]
I dont think we have that concept in moose
Prescribing the displacement on all nodes in a BC is one way to perform the tensile test. Another is to prescribe a force / stress

**Reply by sidharthsarmah** [reply]
Thanks @GiudGiud , as you suggested to use Parsedmaterial instead, I used it but I am getting this error:
```
*** ERROR ***
Invalid function
if(t> 86400, 0, t*3e-3)

in ParsedMaterialHelper.
Syntax error: Unknown identifier
```
```
[BC]
  [./x_load]
    type = ADFunctionDirichletBC
    variable = disp_x
    boundary = 'Load_x'
    function = 'force'
  [../]
[]
[Materials]
  [./force_cond]
    type = ADParsedMaterial
    expression = 'if(t> 86400, 0, t*3e-3)'
    property_name = 'force'
  [../]
[]

**Reply by GiudGiud** [reply]
hello

it's a FunctionDirichletBC so you have to use a Function not a Material
So a ParsedFunction instead of a ParsedMaterial

Guillaume

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud error:
```
*** ERROR ***
Task add_material is not registered to build Function derived objects
```

```
  [./conc_cond]
    type = ADParsedFunction
    expression = 'if(t<86400, 12371.43, 0)'
    property_name = 'conc'
  [../]
```

**Reply by GiudGiud** [reply]
I think Functions go in the Functions block not the Materials block
